model/ftn/w3iogomd.ftn

```diff
@@ -732,15 +734,21 @@
       CASE('IBG')
         I = 1
         J = 7
+      CASE('TAU')
+        I = 1
+        J = 8
+      CASE('RHO')
+        I = 1
+        J = 9
 !/BT4           CASE('D50')
```

Comment:
Instead of changing existing variables I,J numbers, why aren't these just added to the end?

Reply:
The order is not really relevant here. I used this order so that it is similar to the field order in ww3_shel

Reply:
At one point order did matter, in the sense that the (I,J) array was used other places to signify a particular variable.  Perhaps we don't have any more of that?

Reply:
Yes, the order matters in other parts of the code, that this has been taken care of accordingly, for example in ww3_ounf:
https://github.com/ukmo-waves/WW3/blob/73ea7e11167863ef6f5e679351d1c9b929096054/model/ftn/ww3_ounf.ftn#L1106-L1112

Reply:
I still think it's a bad idea to switch existing parameters to new values instead of adding the new values as the new ones.  It's so easy to miss somewhere in the code where these parameters are being changed.  For example, did ww3_grib get looked at?  I also don't think there's a systematic way to ensure that everywhere in the code switched these values; so I think the best practice would be to add the new variables.   What are other's thoughts on this: @aliabdolali @mickaelaccensi @ukmo-ccbunney @thesser1 or anyone else?

Reply:
@JessicaMeixner-NOAA I can understand see Juan's argument here. The problem is that the number of available parameters in group 1 changes depending on the existence of the `IS2` and `BT4` switches. Currently, Group 1 has 8 switches (ignoring Juan's additional RHO output), but if `BT4` is enabled it will have 9, if `IS2` is enabled it will have 10 and is `SETUP` is enabled it will have 12! Placing a new output that is not dependent on a switch before these D50, IC1 and IC5 outputs makes sense programmatically as the total number of switches will then always be known.  I think this makes more sense if you look at lines 674 - 677 in w3odatmd (NOGE is the number of outputs for a group):

```
           NOGE(1) = 9
!/BT4      NOGE(1) = 10 
!/IS2      NOGE(1) = 12
!/SETUP    NOGE(1) = 13
```
Adding a new output on the end of group 1 would force you to specify the flags for D50, IC1 and IC5 even if the BT4, IS2 and SETUP switches are not enabled.

Sorry - that was a bit verbose - I hope that makes sense!

Adding a new output after